### PR TITLE
Add convenience methods for identifying CSS related messages

### PR DESCRIPTION
**Pull request by kingthorin**
- HttpRequestHeader > Add `isCss()` method, refactor `isImage()` method, add private `isSpecificyType(Pattern)` method.
- HttpResponseHeader > Add `isCss()` method.

Related to zaproxy/zaproxy#6288

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by thc202** [comment]
Could you add tests?

**Comment by kingthorin** [comment]
Sure :)

**Comment by kingthorin** [comment]
Now with Unit Tests.

File: zap/src/test/java/org/parosproxy/paros/network/HttpResponseHeaderUnitTest.java [hunk 1]
```diff
@@ -126,4 +130,43 @@ public void shouldSetEmptyReasonPhraseIfNull() throws Exception {
...
+    @ValueSource(
+            strings = {
+                "text/html", // Text but not css
+                "image/png", // Not text or css
+                "teXt/hTmL", // Mixed case
+                "text/html; charset=UTF-8", // Expected charset
+                "text/html;charset=UTF-8", // Charset without space
+                "charset=UTF-8; text/html" // Charset first
+            })
+    public void isCssShouldReturnFalseWhenContentTypeIndicatesCss(String contentType) {
```
Comment (psiinon) [inline zap/src/test/java/org/parosproxy/paros/network/HttpResponseHeaderUnitTest.java:144]: ..WhenContentTypeDoesnNotIndicateCss ?

File: zap/src/test/java/org/parosproxy/paros/network/HttpRequestHeaderUnitTest.java [hunk 2]
```diff
@@ -183,6 +188,52 @@ public void shouldReplaceAnyCookieHeaderWhenSettingCookieParams() {
...
+                "http://example.org/css/styles.css?foo=bar&thing=.css", // In path, ignoring params
+            })
+    public void isCssShouldReturnTrueWhenUrlIndicatesCss(String url) {
+        // Given
+        HttpRequestHeader reqHeader = createRequestHeader(url);
+        // When / Then
+        assertTrue(reqHeader.isCss());
+    }
+
+    private HttpRequestHeader createRequestHeader(String url) {
```
Comment (thc202) [inline zap/src/test/java/org/parosproxy/paros/network/HttpRequestHeaderUnitTest.java:227]: Can be static (same comment in the other test class).

File: zap/src/test/java/org/parosproxy/paros/network/HttpResponseHeaderUnitTest.java (hunk 1 above)
Comment (kingthorin) [reply zap/src/test/java/org/parosproxy/paros/network/HttpResponseHeaderUnitTest.java:144]: Oops, paste fail ...

**Comment by kingthorin** [comment]
Fixed

File: zap/src/test/java/org/parosproxy/paros/network/HttpResponseHeaderUnitTest.java (hunk 1 above)
Comment (thc202) [reply zap/src/test/java/org/parosproxy/paros/network/HttpResponseHeaderUnitTest.java]: It was added one `n` before `Not`.

Comment (kingthorin) [reply zap/src/test/java/org/parosproxy/paros/network/HttpResponseHeaderUnitTest.java]: :( we did it to each other, I copied psiinon's text above for the adjustment :(

Fixed the fix :)

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thanks!

**Review by psiinon — APPROVED** [review]

**Pull request merged**